packages/@aws-cdk/aws-appmesh/lib/shared-interfaces.ts

```diff
@@ -143,28 +143,138 @@ class FileAccessLog extends AccessLog {
    * @default - no file based access logging
    */
   public readonly filePath: string;
+  private readonly virtualNodeLoggingFormat?: CfnVirtualNode.LoggingFormatProperty;
+  private readonly virtualGatewayLoggingFormat?: CfnVirtualGateway.LoggingFormatProperty;
 
-  constructor(filePath: string) {
+  constructor(filePath: string, loggingFormat?: LoggingFormat) {
     super();
     this.filePath = filePath;
+    // For now we have the same setting for Virtual Gateway and Virtual Nodes
+    this.virtualGatewayLoggingFormat = loggingFormat?.bind().formatConfig;
+    this.virtualNodeLoggingFormat = loggingFormat?.bind().formatConfig;
   }
 
   public bind(_scope: Construct): AccessLogConfig {
     return {
       virtualNodeAccessLog: {
         file: {
           path: this.filePath,
+          format: this.virtualNodeLoggingFormat,
         },
       },
       virtualGatewayAccessLog: {
         file: {
           path: this.filePath,
+          format: this.virtualGatewayLoggingFormat,
         },
       },
     };
   }
 }
 
+/**
+ * All Properties for Envoy Access Logging Format for mesh endpoints
+ */
+export interface LoggingFormatConfig {
+  /**
+   * CFN configuration for Access Logging Format
+   *
+   * @default - no access logging format
+   */
+  readonly formatConfig?: CfnVirtualNode.LoggingFormatProperty;
```

Comment:
Please do not expose these `Cfn` types to the user in the L2. Since this looks like something that's only used for the `bind()` method, we may not need to export this. If you absolutely must export this exact type, please copy the members to a new type.